shell/browser/mac/electron_application_delegate.mm

```diff
@@ -157,4 +157,31 @@ - (IBAction)newWindowForTab:(id)sender {
   electron::Browser::Get()->NewWindowForTab();
 }
 
+- (void)application:(NSApplication*)application
+    didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceToken {
+  // https://stackoverflow.com/a/16411517
+  const char* tokenData = static_cast<const char*>([deviceToken bytes]);
+  NSMutableString* tokenString = [NSMutableString string];
+  for (NSUInteger i = 0; i < [deviceToken length]; i++) {
+    [tokenString appendFormat:@"%02.2hhX", tokenData[i]];
+  }
+  electron::Browser::Get()->DidRegisterForAPNSNotificationsWithDeviceToken(
+      base::SysNSStringToUTF8(tokenString));
```

Comment:
Instead of directing this through `electron::Browser`, expose the `PushNotifications` singleton and access it directly. i.e.
```suggestion
  electron::api::PushNotifications::Get()->DidRegisterForAPNSNotificationsWithDeviceToken(
      base::SysNSStringToUTF8(tokenString));
```

along with the associated machinery for `PushNotifications` to be a singleton with a static `Get()` method.

There are a couple of things to consider here. In particular, the `PushNotifications` singleton is currently not created until the Node environment boots up (during the `Initialize` function), whereas `Browser` is always available. We need to avoid calling functions on an uninstantiated object, so I'd recommend:

1. When the `Initialize` function runs and the `PushNotifications` object is created, store a pointer to it in a global.
2. When the `PushNotifications` object is deleted, clear the global (set it to `nullptr`).
3. Create a new `PushNotifications::Get()` method which returns the (possibly null) value of the pointer.
4. When we access it here, first check that it's non-null.

This should also let you remove the new `BrowserObserver` methods.

Basically, we can do all this without involving the `Browser` object!